### PR TITLE
PI-1084 Implement automated functional test for updating Supplier Assessment Appointment in Refer and Monitor and verifying changes in Delius

### PR DESCRIPTION
**Pull request by smalepati99**
Refactor and Add Test Coverage for NSI Contact Verification

This commit introduces significant changes and additions to improve the verification of NSI (Non Statutory Intervention) contact details in the Delius system. The following updates have been made:

- Addition of **verifyNSIContactInDelius:** This function verifies the NSI contact details in Delius. It checks various aspects, including the appointment type, outcome, appointment date and time, attendance, and compliance. The function utilizes specific locators to extract the relevant information from the Delius page and compares it with the expected values.
- Addition of **navigateToNSIContactDetails**: This function navigates to the page in Delius where the NSI contact details are displayed. It follows a sequence of steps to reach the desired page and ensures the correct navigation flow.
- Refactored **createSupplierAssessmentAppointment**: to return the initial appointment date so that this R&M date can be formatted to Delius format
- **Addition of Test Case:** A new test case has been added to the existing test suite to cover the scenario of rescheduling a supplier assessment appointment. The test case ensures that the changes made in the Refer and Monitor system, including the updated appointment outcome, original attendance status, and the creation of a new appointment with the updated date and time, are accurately reflected in the Delius system.

The test case follows these steps:

1. Creation of an offender, community event, and requirement for the event.
2. Creation of an initial supplier assessment appointment in the Refer and Monitor system.
3. Verification of the initial supplier assessment appointment in Delius.
4. Rescheduling of the supplier assessment appointment in the Refer and Monitor system.
5. Verification of both the initial and rescheduled supplier assessment appointments in Delius, ensuring the correct reflection of changes.

These additions enhance the test coverage for rescheduling supplier assessment appointments. They also ensure the accuracy of data synchronization between the Refer and Monitor and Delius systems. 

This commit addresses the aforementioned additions, providing reliable and comprehensive NSI contact verification functionality.

### CODE REVIEW

File: tests/refer-and-monitor-and-delius/create-referral-functional.spec.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import { Tomorrow } from '../../steps/delius/utils/date-time.js'
+import { createAndAssignReferral } from './create-referral.spec.js'
+test.beforeEach(async ({ page }) => {
+    await loginDelius(page)
+})
+
+test('Update Supplier Assessment Appointment in Refer and Monitor to future date and Verify Changes in Delius', async ({
+    page,
+}) => {
+    test.slow()
```
Comment (marcus-bcl) [inline tests/refer-and-monitor-and-delius/create-referral-functional.spec.ts]: Does this take longer than 3 minutes to run, the other R&M tests don't need this do they?

File: tests/refer-and-monitor-and-delius/create-referral-functional.spec.ts
Comment (marcus-bcl) [inline tests/refer-and-monitor-and-delius/create-referral-functional.spec.ts:1]: Could we give this a more meaningful filename?  E.g. `reschedule-appointment.spec.ts`?

**Review by marcus-bcl — DISMISSED** [review]

File: tests/refer-and-monitor-and-delius/create-referral-functional.spec.ts (hunk 1 above)
Comment (smalepati99) [reply tests/refer-and-monitor-and-delius/create-referral-functional.spec.ts]: we noticed that that R&M application runs on a single pod at times, which is causing the tests to exceed the default execution time of 18000 ms. To address this issue and prevent test failures due to premature termination, I have added the test.slow() function to increase the total execution time of the tests, allowing them to complete successfully even when the R&M application is running on a single pod.

File: tests/refer-and-monitor-and-delius/create-referral-functional.spec.ts
Comment (smalepati99) [reply tests/refer-and-monitor-and-delius/create-referral-functional.spec.ts:1]: Agreed. I will rename the spec file and push the changes

Comment (smalepati99) [reply tests/refer-and-monitor-and-delius/create-referral-functional.spec.ts:1]: Done

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**